--- a/python/private/pypi/extension.bzl	2025-12-31 13:31:49.627449122 -0800
+++ b/python/private/pypi/extension.bzl	2025-12-31 13:31:49.627449122 -0800
@@ -146,6 +146,10 @@
         whl_group_mapping = {}
         requirement_cycles = {}
 
+    extra_pip_args = pip_attr.extra_pip_args
+    if module_ctx.os.environ.get("AOS_RUNNING_IN_CI") == "1" and pip_attr.extra_pip_args_ci:
+        extra_pip_args = pip_attr.extra_pip_args_ci
+
     requirements_by_platform = parse_requirements(
         module_ctx,
         requirements_by_platform = requirements_files_by_platform(
@@ -154,14 +158,14 @@
             requirements_lock = pip_attr.requirements_lock,
             requirements_osx = pip_attr.requirements_darwin,
             requirements_windows = pip_attr.requirements_windows,
-            extra_pip_args = pip_attr.extra_pip_args,
+            extra_pip_args = extra_pip_args,
             python_version = full_version(
                 version = pip_attr.python_version,
                 minor_mapping = minor_mapping,
             ),
             logger = logger,
         ),
-        extra_pip_args = pip_attr.extra_pip_args,
+        extra_pip_args = extra_pip_args,
         get_index_urls = get_index_urls,
         # NOTE @aignas 2024-08-02: , we will execute any interpreter that we find either
         # in the PATH or if specified as a label. We will configure the env
@@ -646,6 +650,7 @@
         A dict of attributes.
     """
     attrs = dict({
+        "extra_pip_args_ci": attr.string_list(doc = "CI args override"),
         "experimental_extra_index_urls": attr.string_list(
             doc = """\
 The extra index URLs to use for downloading wheels using bazel downloader.
@@ -910,6 +915,7 @@
 """,
         ),
     },
+    environ = ["AOS_RUNNING_IN_CI"],
 )
 
 def _whl_mods_repo_impl(rctx):
