diff --git a/absl/container/flat_hash_map_test.cc b/absl/container/flat_hash_map_test.cc
index 5c83c941..c6dc8ee0 100644
--- a/absl/container/flat_hash_map_test.cc
+++ b/absl/container/flat_hash_map_test.cc
@@ -116,7 +116,7 @@ TEST(FlatHashMap, StandardLayout) {
 
 TEST(FlatHashMap, Relocatability) {
   static_assert(absl::is_trivially_relocatable<int>::value);
-#if ABSL_INTERNAL_CPLUSPLUS_LANG <= 202002L
+#if ABSL_INTERNAL_CPLUSPLUS_LANG <= 202002L && !defined(__FRC_ROBORIO__)
   // std::pair is not trivially copyable in C++23 in some standard
   // library versions.
   // See https://github.com/llvm/llvm-project/pull/95444 for instance.
