diff --git a/bazel/copts.bzl b/bazel/copts.bzl
index 6aff817649..bc4b28890f 100644
--- a/bazel/copts.bzl
+++ b/bazel/copts.bzl
@@ -165,5 +165,5 @@ GRPC_LLVM_WINDOWS_WARNING_FLAGS = GRPC_LLVM_WARNING_FLAGS + [
 GRPC_DEFAULT_COPTS = select({
     "//:use_strict_warning": GRPC_LLVM_WARNING_FLAGS + ["-DUSE_STRICT_WARNING=1"],
     "//:use_strict_warning_windows": GRPC_LLVM_WINDOWS_WARNING_FLAGS + ["-DUSE_STRICT_WARNING=1"],
-    "//conditions:default": [],
+    "//conditions:default": ["-Wno-missing-constinit"],
 })
diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index 4a8e18ce17..50c7c753b4 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -64,9 +64,9 @@ def grpc_extra_deps(ignore_version_differences = False):
 
     api_dependencies()
 
-    go_rules_dependencies()
-    go_register_toolchains(version = "1.22.5")
-    gazelle_dependencies()
+    #go_rules_dependencies()
+    #go_register_toolchains(version = "1.22.5")
+    #gazelle_dependencies()
 
     # Pull-in the go 3rd party dependencies for protoc_gen_validate, which is
     # needed for building C++ xDS protos
